readme.txt

```diff
@@ -11,3 +11,38 @@ Insanely powerful admin search, powered by Elasticsearch.
 
 == Description ==
 
+Adds a powerful faceted search page to the admin area to search all posts across all post types.
+
+Optionally, this plugin can also replace core's post search forms throughout the entire admin area to leverage Elasticsearch for faster, more relevant search results.
+
+== Prerequisites ==
+
+* Elasticsearch server
+* Elasticsearch plugin/adapter (e.g. [SearchPress](https://github.com/alleyinteractive/searchpress)
+* [ES_WP_Query](https://github.com/alleyinteractive/es-wp-query) if you want to use the optional core search replacement.
+
+== Installation ==
+
+ES Admin is designed to be able to work with any Elasticsearch index, regardless of how the data is mapped. In order to accomplish this, ES Admin requires you to register an "Adapter", which maps appropriate post fields (e.g. `post_content`) to the Elasticsearch index path (e.g. `post_content.analyzed`). ES Admin currently comes with three adapters: [SearchPress](https://github.com/alleyinteractive/searchpress), [WordPress.com](https://vip-svn.wordpress.com/plugins/wpcom-elasticsearch/wpcom-elasticsearch.php) (used by WordPress.com VIP sites hosted on the main WordPress.com platform), and a generic adapter primarily used for unit testing. You aren't limited to using one of the included adapters, you can very easily build your own to work with your Elasticsearch plugin of choice by extending the `ES_Admin\Adapters\Adapter` abstract class. See the [Generic Adapter](https://github.com/alleyinteractive/es-admin/blob/master/lib/adapters/class-generic.php) as an example to create your own.
+
+Once you have your adapter, you register it with ES Admin using the `es_admin_adapter` filter. This filter expects a class name in return, which must be an `ES_Admin\Adapters\Adapter` extension. Here's an example using this filter with the SearchPress adapter:
+
+```php
+add_filter( 'es_admin_adapter', function( $adapter ) {
```

Comment:
Non-blocking, but you don't need to add the `$adapter` variable here since it is not being used.